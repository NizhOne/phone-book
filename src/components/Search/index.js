import { connect } from 'react-redux';
import SearchComponent from './Search';
import { search } from '../../actions';

const mapDispatchToProps = (dispatch) => {
    return {
        onSearchChange: (searchString) => {
            dispatch(search(searchString))
        }
    }
};

const mapStateToProps = (state) => {
    return {
        searchString: state.searchContacts
    }
};

const Search = connect(
    mapStateToProps,
    mapDispatchToProps
)(SearchComponent);

export default Search;