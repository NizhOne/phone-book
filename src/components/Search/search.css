.search {
    width: 220px;
    height: 28px;
    position: relative;
    margin: 20px;
}
.search__input {
    width: 170px;
    height: 100%;
    border: 1px solid #BDBDBD;
    border-radius: 4px;
    padding-left: 20px;
    padding-right: 30px;
    background-color: #ffffff;
    background-image: url('../../assets/icons/search.svg');
    background-size: 15px 15px;
    background-repeat: no-repeat;
    background-position: right 10px center;
}
.search input::placeholder {
    font-size: 12px;
}
.clickable {
    height: 28px;
    width: 30px;
    position: absolute;
    right: 0;
    top: 0;
    cursor: pointer;
}