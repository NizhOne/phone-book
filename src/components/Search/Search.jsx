import React from 'react';
import PropTypes from 'prop-types';

import './search.css';

const SearchComponent = ({onSearchChange, value}) => (
    <div className={`search`}>
        <input
            className={`search__input`}
            placeholder={'Search'}
            onChange={({target: {value}}) => onSearchChange(value)}
            value={value}
            type={`search`}
        />
        <div className={`clickable`}/>
    </div>
);

SearchComponent.propTypes = {
    onSearchChange: PropTypes.func.isRequired,
    value: PropTypes.string
};

export default SearchComponent;