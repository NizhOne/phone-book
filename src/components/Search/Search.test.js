import React from 'react';
import Search from './Search';
import renderer from 'react-test-renderer';


test('test', () => {
    const component = renderer.create(
        <Search onSearchChange={() => true} value={'hello'} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});