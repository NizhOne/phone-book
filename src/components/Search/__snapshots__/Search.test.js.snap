// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1`] = `
<div
  className="search"
>
  <input
    className="search__input"
    onChange={[Function]}
    placeholder="Search"
    type="search"
    value="hello"
  />
  <div
    className="clickable"
  />
</div>
`;
